Mafia can be played on your PDA

JohnFulpWillard opened the pull request:
## About The Pull Request

Mafia is now friggin playable from the PDA, I also changed other stuff too

- You can't use abilities on dead people if you're not supposed to (cant kill the same person over and over)
- Changelings cant kill other Changelings
- Changelings can now only talk to eachother at night, rather than using :j
- Everyone starts spawned in the center of the map, since people playing on PDA can't move their characters. This is so everyone can hear PDA users in person, as I don't want the chat log to be mandatory.

To do this, all messages you are meant to be able to see, is now logged for you to see in your Mafia panel. This essentially means that people playing through the PDA get a downgraded version of it, but I don't know how much larger I want this UI to be.

Playing Mafia through the PDA will not tell you of other players ahead of time when signing up (as it shows ckeys + pdas), but they can see the names in-game. Unfortunately this means we'll have to remove your customization coming with you, to prevent using it to tell who is dead in round.

Things I am missing
- Program overlays on PDA/Laptop/Computer
- Icon for the app's header while a game is active

I'm not a spriter and can't make either of these

This is the new UI
![image](https://github.com/tgstation/tgstation/assets/53777086/7cf503d9-b2e2-4127-874a-acad6425d649)

I also fixed alert calls for PDAs and stuff
![image](https://github.com/tgstation/tgstation/assets/53777086/e09b2e5e-b9e7-43ae-9273-c168e9c8e642)

and removed the X at the top on computers since they had no battery
![image](https://github.com/tgstation/tgstation/assets/53777086/d3dd8307-805c-4aba-be5e-4c24a0bdcb91)

Looks a little better now hopefully 👍 

## Why It's Good For The Game

- The current Arcade app sucks, and is a solo game. This is much more entertaining and you can talk to others in it, which is swag as fuck.
- There's a larger potential playerbase for the Minigame making it more likely to be played.
- Sets groundwork for a better version of https://github.com/tgstation/tgstation/pull/75879
- Adds more suspense and teamwork in the minigame.

## Changelog

:cl: JohnFulpWillard, sprites by CoiledLamb
add: You can now play Mafia on your PDA.
balance: Mafia changelings can now only talk to eachother during the night.
fix: Mafia abilities can't be repeatedly used on people.
/:cl:

JohnFulpWillard committed:
a lot of temporary stuff atm

JohnFulpWillard committed:
Rest of the stuff

JohnFulpWillard committed:
Hide some info from observers

JohnFulpWillard committed:
Update launch.json

JohnFulpWillard committed:
Update launch.json

bawbbawbins commented:
jhony wanna paly bloxburg later today

JohnFulpWillard commented:
i dont have a roblox account

JohnFulpWillard committed:
remove a testing thing

JohnFulpWillard committed:
Everyone starts at the center

Y0SH1M4S73R commented:
> Unfortunately this means we'll have to remove your customization coming with you, to prevent using it to tell who is dead in round.

Would it be feasible to give ghost players a pseudonym that appears when they are referenced on the PDA, or would it be too difficult to catch instances where part of a character's name is used in speech?

JohnFulpWillard commented:
> Would it be feasible to give ghost players a pseudonym that appears when they are referenced on the PDA, or would it be too difficult to catch instances where part of a character's name is used in speech?

I think it would ruin part of the experience, calling someone by their first/last/short names would not be caught and then you'd have confusion of 2 groups of people calling someone by 2 different names.

JohnFulpWillard committed:
don't spam & don't bring prefs

JohnFulpWillard committed:
Fixes the UI staying after the game is over & better pda stuff

JohnFulpWillard committed:
AAAA EFFECTS IN DESTROY

Dawnseer commented:
Alright but what if I got someone else's PDA, and I use both my own PDA and theirs to play. Will I be able to rig the game?

JohnFulpWillard commented:
> Alright but what if I got someone else's PDA, and I use both my own PDA and theirs to play. Will I be able to rig the game?

Yea

Dawnseer commented:
Sweet. I am now imagining the HoP and some morgue looter playing a full game of mafia but it's just those two playing all the roles.

JohnFulpWillard commented:
Names are all anonymous so it'll be harder to figure out who is actually playing, outside of the alert that a new game started (which I could remove)- and that's just life I suppose.

Dawnseer commented:
Just gotta normalize centcomm sending deathsquads after people who cheat at mafia. Problem solved, and fun for all parties involved. Probably.

EricZilla commented:
I cannot beleive we can metacomm IC now

JohnFulpWillard commented:
preferably you would not

JohnFulpWillard committed:
Better logs & sanitizing

JohnFulpWillard commented:
I'm gonna have to close this until I got sprites for it

Time-Green commented:
😢

JohnFulpWillard committed:
Adds & fixes icons for stuff

JohnFulpWillard committed:
Merge branch 'master' into mafia-pda

Ghommie commented on code/modules/mafia/controller.dm:
```diff
@@ -584,21 +608,28 @@ GLOBAL_LIST_INIT(mafia_role_by_alignment, setup_mafia_role_by_alignment())
-	if(user.client?.holder)
+	if(usr.client?.holder)
 		data["admin_controls"] = TRUE //show admin buttons to start/setup/stop
+	data["is_observer"] = !!isobserver(user)
```
`istype()` already return booleans.

Ghommie commented on code/modules/mafia/controller.dm:
```diff
@@ -619,18 +650,25 @@ GLOBAL_LIST_INIT(mafia_role_by_alignment, setup_mafia_role_by_alignment())
 		data["user_notes"] = user_role.written_notes
+		data["messages"] = list()
+		var/list/ui_messages = list()
+		for(var/i = user_role.role_messages.len to 1 step -1)
```
A part of me wants to suggest `reverseList`, but it's no big deal really.

Ghommie reviewed (changes requested):
You need to make the `award_role` proc compatible with modular computer / PDA players, if you haven't, otherwise lgtm except one and a half nitpicks.

JohnFulpWillard committed:
You did this to me.........

I looked around in the code and saw shit I really wanted to change so I ended up changing them because I friggin can ok?
